### PR TITLE
Add in filters for Deals section + add in latest deals

### PR DESCRIPTION
**Pull request by jvmi7**
### TL;DR

Add new promotions to the PromotionsSection in the DApp Console. Includes Privy, Bware, and Sherlock deals.

### Details

- Added new promotions metadata for Privy, Bware, and Sherlock
- Updated PromotionsSection to display new promotions based on category filter

https://github.com/ethereum-optimism/ecosystem/assets/10838132/22269ca2-24f1-470d-bbe6-4f6aa9b734b5

### CODE REVIEW

**Comment by jvmi7** [comment]
* **#307** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jvmi7 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dapp-console/app/console/useDialogContent.tsx [hunk 1]
```diff
@@ -170,6 +173,33 @@ const useDialogContent = () => {
...
+  }, [authenticated, login])
+
+  const sherlockContent = useMemo(() => {
+    return renderDialog({
+      ...sherlockMetadata,
+      primaryButton: !authenticated
+        ? loginButton('Sign in to apply', sherlockMetadata.label)
+        : sherlockMetadata.primaryButton,
+    })
+  }, [authenticated, login])
```
Comment (jakim929) [inline apps/dapp-console/app/console/useDialogContent.tsx:201]: nit: might be worth creating a helper function/hook since this pattern is repeated

**Review by jakim929 — APPROVED** [review]
looks good! one optional nit

**Comment by shaunop** [comment]
Agree, looks good @jvmi7. Another small nit, let's tighten up the spacing between the pills. Let's try 8pt.

**Pull request merged**